compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2008,28 +2008,63 @@ boolean checkTypeAndVarCountConsistency(BLangTupleVariable varNode, BTupleType t
                 case TypeTags.TUPLE:
                     tupleTypeNode = (BTupleType) varNode.type;
                     break;
+                case TypeTags.ARRAY:
+                    List<BType> tupleTypes = new ArrayList<>();
+                    BArrayType arrayType = (BArrayType) varNode.type;
+                    for (int i = 0; i < arrayType.size; i++) {
+                        tupleTypes.add(arrayType.eType);
+                    }
+                    tupleTypeNode = new BTupleType(tupleTypes);
```

Comment:
Checked the feasibility of removing this. According to my understanding, I think we'll have to add many checks to places where we have used the created tuple if we are not creating a new tuple here since all other cases also create a tuple and the rest of the logic is written catering to that. Shall we create an issue for this and revisit it if possible?

Reply:
Created an issue. https://github.com/ballerina-platform/ballerina-lang/issues/28838